### PR TITLE
Refactor CI

### PR DESCRIPTION
**Pull request by Zerpet**
**Note to reviewers:** remember to look at the commits in this PR and consider if they can be squashed

## Summary Of Changes

- **Do not notify failed tests in PR**
- **Add a test-ci rule branch**
- **Split image building into different jobs**
- **Install tools as part of unit test**
- **Add permissions for steps using GCP**
- **[ci] Use local builds in other jobs**
- **[ci] Refactor build-test workflow**
- **Fix examples**
- **Update CMCTL binary**
- **Use github token in carvel-setup action**
- **Checkout code before setup GO**
- **Re-enable GChat triggers**

## Additional Context

We are moving away from the massive CI image (~1GB) with a catch-all for tools, into a actions based
on small simple steps that setup the worker as we need to, and rely on the worker cache.

Another important step is moving away from our GCP project for system tests, into local system tests
using KinD. This is important because it relieves the project from depending on sponsored infrastructure,
and it gives more independence to run its test suites.

There's a important fix to the examples job, as it was always passing earlier. Now, any failure in the
examples will also fail the workflow. In PRs, we only dry-run the examples, in order to verify they have
a correct syntax. Once merged, the examples will run in KinD. The examples job is the heaviest, taking
about 17 minutes to run.

### CODE REVIEW

File: docs/examples/federation-over-tls/README.md
[the remark this answers is not in the document]
Comment (MirahImage) [reply docs/examples/federation-over-tls/README.md:1]: That makes a lot of sense, I agree. I just didn't see it stated in the summary so wanted to make sure it was an intentional change.

**Comment by Zerpet** [comment]
@PujaVad @DanielePalaia would you have some bandwidth to review?
I'd like to merge this PR this week, as it is blocking other PRs.

**Comment by DanielePalaia** [comment]
Hey @Zerpet it looks good to me, I noticed that in the test_upgrade we are still using the GKE cluster. I was wondering if we are planning to migrate this one in a next step or it was for some limitations encountered.

**Comment by Zerpet** [comment]
I left it out because I suspect we may have a resource limitation in Actions. The upgrade tests use a 3-node rabbit. Those were not quite deploying when I did this refactor. We can try a couple things to make those tests work in KinD. I would prefer to do that in a separate PR.

**Pull request merged**